### ISSUE TITLE
[ArrowIO] - Old metadata version not supported

### ISSUE THREAD

**Issue by nmatare** [tensorflow/io#806]
While attempting to try out Arrow integration, I run into the below error:

**tensorflow.python.framework.errors_impl.InternalError: Invalid: Old metadata version not supported**

The below reproduces the error:

```
import numpy as np
import pandas as pd

import tensorflow_io
from tensorflow_io import arrow

assert np.__version__ == '1.18.1'
assert pd.__version__ == '1.0.1'
assert tensorflow_io.__version__ == '0.11.0'

data = {'label': np.random.binomial(1, 0.5, 10)}
data['x0'] = np.random.randn(10) + 5 * data['label']
data['x1'] = np.random.randn(10) + 5 * data['label']
df = pd.DataFrame(data)

ds = tensorflow_io.arrow.ArrowDataset.from_pandas(df)
next(iter(ds))
```

And the stacktrace:

> next(iter(ds))
2020-02-24 09:44:25.636894: W tensorflow/core/framework/op_kernel.cc:1655] OP_REQUIRES failed at iterator_ops.cc:941 : Internal: Invalid: Old metadata version not supported
Traceback (most recent call last):
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/eager/context.py", line 1897, in execution_mode
    yield
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/data/ops/iterator_ops.py", line 659, in _next_internal
    output_shapes=self._flat_output_shapes)
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/ops/gen_dataset_ops.py", line 2479, in iterator_get_next_sync
    _ops.raise_from_not_ok_status(e, name)
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 6606, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InternalError: Invalid: Old metadata version not supported [Op:IteratorGetNextSync]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/data/ops/iterator_ops.py", line 630, in __next__
    return self.next()
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/data/ops/iterator_ops.py", line 674, in next
    return self._next_internal()
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/data/ops/iterator_ops.py", line 665, in _next_internal
    return structure.from_compatible_tensor_list(self._element_spec, ret)
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/eager/context.py", line 1900, in execution_mode
    executor_new.wait()
  File "/home/johnconnor/miniconda3/envs/deeptick/lib/python3.7/site-packages/tensorflow_core/python/eager/executor.py", line 67, in wait
    pywrap_tensorflow.TFE_ExecutorWaitForAllPendingNodes(self._handle)
tensorflow.python.framework.errors_impl.InternalError: Invalid: Old metadata version not supported

**Comment by BryanCutler** [comment]
Thanks for the report @nmatare.  Could you confirm which version of pyarrow you have installed?
`python -c "import pyarrow; print(pyarrow.__version__)"`

There was a change in the binary format in Arrow v0.15.0, and TensorFlow I/O v0.11.0 was built with Arrow v0.14.1. So if you are using pyarrow 0.16.0, it seems that it causes an invalid message version, which results in the error you see.  There are a couple options to work around this:

1. You can set an env var for pyarrow to use the old binary format :
`$ export ARROW_PRE_0_15_IPC_FORMAT=1`

2. You could downgrade your pyarrow version to 0.14.1

3. We recently upgraded Arrow to v0.16.0 in TensorFlow I/O and this wouldn't be a problem, so you could build from master until the next release.

I hope that helps, let me know if it fixes your issue.

**Comment by nmatare** [comment]
Thanks for the explanation and response @BryanCutler. 

I can confirm that the issue is solved when setting the environment variable `ARROW_PRE_0_15_IPC_FORMAT=1`. I was running pyarrow 0.15.1.

Alternatively, I checked with pyarrow 0.14.1 and that indeed works as well.

Thanks again.

**Comment by yongtang** [comment]
Added an issue #813 to capture the tensorflow-io 0.12.0 release tracking.

**Comment by BryanCutler** [comment]
Thanks @nmatare , tensorflow-io just released 0.12.0, which will solve this problem also. I'll go ahead and close this.
